src/Controllers/ContentController.php

```diff
@@ -2225,4 +2258,122 @@ public function report(Request $request)
+        }
+        foreach ($types as $type) {
+            $type = $this->getContentTypeForMetaData($type);
+            if (array_key_exists($type, config('railcontent.cataloguesMetadata.' . $brand))) {
+                $catalogMetaAllowableFilters = config('railcontent.cataloguesMetadata.' . $brand . '.' . $type . '.allowableFiltersMobile', ContentRepository::$catalogMetaAllowableFilters);
+            }
+        }
+        if (count($types) > 1) {
+            $catalogMetaAllowableFilters = config('railcontent.cataloguesMetadata.' . $brand . '.all.allowableFiltersMobile', ContentRepository::$catalogMetaAllowableFilters);
+        }
+        ContentRepository::$catalogMetaAllowableFilters = $catalogMetaAllowableFilters;
+
+        $requiredFields = $request->get('required_fields', []);
+        $requiredFields = array_merge($requiredFields, ['style,' . $genre]);
+        if ($request->has('title')) {
+            $requiredFields = array_merge($requiredFields, ['title,%'.$request->get('title').'%,string,like']);
+        }
+
+        $lessons = $this->contentService->getFiltered( $request->get('page', 1),
+                                                              $request->get('limit', 12),
+                                                              $request->get('sort', '-popularity'),
+                                                              $types,
+                                                              $request->get('slug_hierarchy', []),
+                                                              $request->get('required_parent_ids', []),
+                                                              $requiredFields,
+                                                              $request->get('included_fields', []),
+                                                              $request->get('required_user_states', []),
+                                                              $request->get('included_user_states', []),
+                                                              $request->get('with_filters', true));
+
+        $content = new ContentEntity();
+        $content['name'] = $genre;
+        $content['type'] = 'style';
+        $content['thumbnail_url'] = config('railcontent.avatar_style')[$genre] ?? config('railcontent.default_avatar_style')[config('railcontent.brand', 'drumeo')];
+        $content['lessons'] = $lessons['results'];
+        $content['lessons_filter_options'] = $lessons['filter_options'];
+        if($request->get('with_filters', true)) {
+            $content['lesson_count'] = $lessons['total_results'];
+        }
+        $content['total_results'] = $lessons['total_results'];
+
+        return ResponseService::content($content);
+    }
+
+    public function artistCollectionPage(Request $request)
+    {
+        $types = $request->get('included_types');
+        $artist = $request->get('artist');
```

Comment:
Should we do any sort of validation/safety check here, in case the `artist` isn't set?

Reply:
Added. Thanks!